docs/library/machine.Counter.rst

```diff
@@ -0,0 +1,67 @@
+.. currentmodule:: machine
+.. _machine.Counter:
+
+class Counter-- Pulse Counter
+=============================
+
+This class provides access to hardware-supported pulse counting.
+
+It is currently provided for ports:
+
+  * :ref:`ESP32 <esp32_machine.Counter>`
+  * :ref:`MIMXRT <mimxrt_machine.Counter>`
+
+Minimal example usage::
+
+    from machine import Pin, Counter
+
+    counter = Counter(0, src=Pin(0, mode=Pin.INPUT))  # create Counter object and start to count input pulses
+    counter.init(filter_ns=1000)                      # switch source filtering on
+    value = counter.value(0)                          # get current Counter value, set counter to 0
+    counter.deinit()                                  # turn off the Counter
+
+    print(counter)                                    # show the Counter object properties
+
+Constructor
+-----------
+
+.. class:: Counter(id, src=None, \*, direction=1, filter_ns=0)
```

Comment:
Do not use numeric constants. Define symbolic constants Counter.UP and Counter.DOWN, which allows the numeric value to be hardware port specific.